AutomatedTesting/Gem/PythonTests/Terrain/EditorScripts/TerrainPhysicsCollider_MaterialMapping_Works.py

```diff
@@ -0,0 +1,70 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+
+class Tests:
+    enter_game_mode = (
+        "Entered Game Mode",
+        "Failed to enter Game Mode"
+    )
+    exit_game_mode = (
+        "Exited Game Mode",
+        "Failed to exit Game Mode"
+    )
+
+
+def TerrainPhysicsCollider_MaterialMapping_Works():
+    """
+    Summary:
+    Opens an existing level with a Terrain Layer Spawner split into 2 different surface areas with 2 different Physics
+    Materials applied. Test verifies that physicalized objects properly interact with the specified Physics Material
+    mapped surfaces on terrain.
+    :return: None
+    """
+
+    import azlmbr.legacy.general as general
+    import azlmbr.bus as bus
+    import azlmbr.components as components
+
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+
+    # Open a level with preconfigured terrain with surfaces mapped to Physics Materials
+    helper.init_idle()
+    helper.open_level("Terrain", "TerrainPhysicsCollider_MaterialMapping_Works")
+
+    # Enter Game Mode
+    helper.enter_game_mode(Tests.enter_game_mode)
+
+    # Find the needed entities
+    physics_cube_glass = general.find_game_entity("PhysicsCube_Glass")
+    physics_cube_rubber = general.find_game_entity("PhysicsCube_Rubber")
+
+    # Get the initial positions of each Physics Cube
+    glass_start = components.TransformBus(bus.Event, "GetWorldTranslation", physics_cube_glass)
+    rubber_start = components.TransformBus(bus.Event, "GetWorldTranslation", physics_cube_rubber)
+
+    # Wait a few seconds, re-check positions, and calculate distance traveled
+    general.idle_wait(3.0)
+    glass_finish = components.TransformBus(bus.Event, "GetWorldTranslation", physics_cube_glass)
+    rubber_finish = components.TransformBus(bus.Event, "GetWorldTranslation", physics_cube_rubber)
+    glass_distance = glass_finish.GetDistance(glass_start)
+    rubber_distance = rubber_finish.GetDistance(rubber_start)
+
+    # Report results on distance traveled and validate that the Cube on the glass surface travels further
+    material_interaction = (
+        f"PhysicsCube_Glass traveled {glass_distance:.3f}m, to PhysicsCube_Rubber's {rubber_distance:.3f}m",
+        f"PhysicsCube_Rubber traveled {rubber_distance:.3f}m, greater than PhysicsCube_Glass's {glass_distance:.3f}m"
+    )
+    Report.result(material_interaction, glass_distance > rubber_distance)
```

Comment:
is there an expected minimum difference? floating point variation could result in nearly the same travel being seen as greater than. you could test
glass_distance > (rubber_distance + 0.1)
not sure it matters that much though.

Reply:
https://jenkins-pipeline.agscollab.com/blue/organizations/jenkins/O3DE-LY-FORK/detail/jckand%2FTerrainPhysicsMaterialsTest/2/pipeline/318

Reply:
Test Passed
------------
|  Output  |
------------
Starting test TerrainPhysicsCollider_MaterialMapping_Works...
Test TerrainPhysicsCollider_MaterialMapping_Works finished.
Report:
[SUCCESS] Success: Entered Game Mode
[SUCCESS] Success: PhysicsCube_Glass traveled 12.043m, to PhysicsCube_Rubber's 0.029m
[SUCCESS] Success: Exited Game Mode
Test result:  SUCCESS

Reply:
+1 would be nice if this check was at least a little more stringent, while still being forgiving enough to account for any non-determinism. For example:
rubber_distance < 2.0 
glass_distance > 8.0

This would make it clear that there's a significant difference in the amount the two cubes are expected to travel. Right now, a difference of 0.000001 would still pass.